code/src/pages/MoviePage.js

```diff
@@ -0,0 +1,41 @@
+import React, { useEffect, useState } from 'react';
+import { useParams } from 'react-router-dom';
+
+import Loader from '../components/Loader';
+import Error from '../components/Error';
+import MovieDetail from '../components/MovieDetail';
+import { API_KEY } from '../api.js';
+
+const MoviePage = () => {
+  const [isLoading, setLoading] = useState(true);
+  const [movie, setMovie] = useState([]);
+  const { movieId } = useParams();
+  const MOVIE_URL = `https://api.themoviedb.org/3/movie/${movieId}?api_key=${API_KEY}&language=en-US`;
+
+  useEffect(() => {
+    fetch(MOVIE_URL)
+      .then(result => {
+        if (result.ok) {
+          return result.json();
+        } else {
+          throw new Error('404');
+        }
+      })
+      .then(json => {
+        setMovie(json);
+        setLoading(false);
+      })
+      .catch(() => {
+        window.location.assign('/error');
+      });
+  }, [MOVIE_URL]);
+
+  return (
+    <>
```

Comment:
Over here as well, since `isLoading` is either `true` or `false`, but not those two values at the same time. There is no way your component will return both of those components at the same time